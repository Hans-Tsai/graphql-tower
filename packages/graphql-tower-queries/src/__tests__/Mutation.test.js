import { graphql, GraphQLSchema, GraphQLObjectType, GraphQLID, GraphQLInt } from 'graphql';
import { Mutation } from '../';

describe('Mutation', () => {
  it('snapshot', async () => {
    const MutationNode = class extends Mutation {
      inputFields = {};
      outputFields = {};
    };
    const mutation = new MutationNode();
    expect(mutation).toMatchSnapshot();
  });

  it('with fields', async () => {
    const MutationNode = class extends Mutation {
      inputFields = () => ({
        id: { type: GraphQLID },
        value: { type: GraphQLInt },
      });
      outputFields = () => ({
        id: { type: GraphQLID },
        value: { type: GraphQLInt },
      });
      resolve = (payload, { input }) => input;
    };

    const schema = new GraphQLSchema({
      query: new GraphQLObjectType({ name: 'Query', fields: { mutation: new MutationNode() } }),
    });

    const { data } = await graphql(schema, 'query { mutation ( input: { id: 10 value: 99 }) { id value } }');
    expect(data).toEqual({ mutation: { id: '10', value: 99 } });
  });
});
