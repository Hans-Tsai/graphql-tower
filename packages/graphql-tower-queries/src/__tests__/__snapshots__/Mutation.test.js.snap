// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mutation snapshot 1`] = `
MutationNode {
  "_": Object {
    "afterware": Array [],
    "middleware": Array [],
    "name": undefined,
    "resolve": [Function],
    "subscribe": [Function],
  },
  "args": Object {
    "input": Object {
      "type": "MutationNodeInput!",
    },
  },
  "resolve": [Function],
  "type": "MutationNodePayload",
}
`;
